### PR TITLE
Remove vertex-based stencil from FV.

### LINKED ISSUE #20953: Remove vertex-based interpolation from finite volume.
## Reason
Because it is not used, was not used and I wouldn't recommend anybody using it. It is provably at least first order accurate but even for simple cases it fails to reach second order. Also, its computation time is comparable of the skewness-corrected face interpolation which manages to reach second order accuracy. 

## Design
Delete source and tests.

## Impact
Easier code management, less confusion about face interpolation.

### PR DESCRIPTION
**Pull request by grmnptr**
Closes #20953

### CODE REVIEW

**Review by lindsayad — CHANGES_REQUESTED** [review]
See above

File: test/tests/fvkernels/fv_adapt/tests [hunk 1]
```diff
@@ -27,7 +27,7 @@
     type = PythonUnitTest
     input = 'steady-adapt-run-and-plot.py'
     test_case = TestSteadyAdapt
-    requirement = 'The system shall display first order convergence when there is significant skewness in the mesh due to mismatch between cell levels.'
+    requirement = 'The system shall display second order convergence when there is an insignificant skewness in the mesh due to mismatch between cell levels.'
```
Comment (lindsayad) [inline test/tests/fvkernels/fv_adapt/tests:30]: Hmm, this seems like an overcorrection. The skewness was significant enough in the past that our original interpolation weights gave first order results. So it's not *insignificant*. Maybe change to
```suggestion
    requirement = 'The system shall display second order convergence when there is some skewnewss due to mismatch between cell levels.'
```

File: framework/src/variables/MooseVariableFV.C [hunk 2]
```diff
@@ -1005,10 +943,7 @@ MooseVariableFV<OutputType>::evaluate(const FaceArg & face,
     return getExtrapolatedBoundaryFaceValue(*fi);
   else if (isInternalFace(*fi))
   {
-    if (_face_interp_method == Moose::FV::InterpMethod::VertexBased)
-      return getInternalFaceValue(*fi, face.correct_skewness);
-    else
-      return Moose::FV::interpolate(*this, face);
+    return Moose::FV::interpolate(*this, face);
   }
```
Comment (lindsayad) [inline framework/src/variables/MooseVariableFV.C:947]: Can now remove the enclosing braces

File: framework/src/variables/MooseVariableFV.C [hunk 3]
```diff
@@ -637,22 +591,7 @@ MooseVariableFV<OutputType>::getInternalFaceValue(const FaceInfo & fi,
...
-      denominator += 1. / distance;
-    }
-
-    value = numerator / denominator;
-  }
-  else
-    value = Moose::FV::linearInterpolation(*this, Moose::FV::makeCDFace(fi, correct_skewness));
+  value = Moose::FV::linearInterpolation(*this, Moose::FV::makeCDFace(fi, correct_skewness));
 
   return value;
```
Comment (lindsayad) [inline framework/src/variables/MooseVariableFV.C:596]: Might as well tighten up this white space (e.g. probably just delete the separating new-lines)

**Review by lindsayad — CHANGES_REQUESTED** [review]
I wasn't going to make the small code comments, but since I do think we should reword the requirement, I decided to. Otherwise this is good to merge

File: framework/src/variables/MooseVariableFV.C (hunk 2 above)
Comment (grmnptr) [reply framework/src/variables/MooseVariableFV.C:947]: Addressed!

File: framework/src/variables/MooseVariableFV.C (hunk 3 above)
Comment (grmnptr) [reply framework/src/variables/MooseVariableFV.C:596]: Addressed!

**Review by lindsayad — APPROVED** [review]

**Review by GiudGiud — APPROVED** [review]

**Comment by lindsayad** [comment]
https://civet.inl.gov/job/1495623/ shows the test failure. After the test was switched to `heavy` a long time ago it was not run until the last day(s)

**Comment by lindsayad** [comment]
I also think there's a bug as I would not expect error to increase with refinement. @grmnptr if you're willing to look into it, that would be great. I would focus on code that would have branching relevant to adaptivity such as `loopOverElemFaceInfo` in `FVUtils.h`

**Comment by grmnptr** [comment]
Got it, thanks! I'm on it!

**Comment by grmnptr** [comment]
This should fix it:
https://github.com/idaholab/moose/pull/24241

**Pull request merged**